packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelFactoryProvider.cs

```diff
@@ -151,13 +152,15 @@ private IReadOnlyList<MethodBodyStatement> GetCollectionInitialization(MethodSig
             return [.. statements];
         }
 
-        private static IReadOnlyList<ParameterProvider> GetParameters(ConstructorProvider modelFullConstructor)
+        private static IReadOnlyList<ParameterProvider> GetParameters(
+            ConstructorProvider modelFullConstructor,
+            bool supportsBinaryDataAdditionalProps)
         {
             var modelCtorParams = modelFullConstructor.Signature.Parameters;
             var parameters = new List<ParameterProvider>(modelCtorParams.Count);
             foreach (var param in modelCtorParams)
             {
-                if (param.Name.Equals(AdditionalRawDataParameterName))
+                if (param.Name.Equals(AdditionalBinaryDataParameterName) && !supportsBinaryDataAdditionalProps)
```

Comment:
Curious why we need the supportsBinaryDataAdditionalProps. When would we have a BinaryData AP parameter in the ctor if the model doesn't support it?